File: Robot co-ordinates.py
"""
The initial x and y co-ordinate values of a Robot are passed as the input.
The rest of the input values are the directions in which the Robot moves along with the distance in that direction.
The directions are denoted by N, E, S, W for North, East, South and West.
The program should print the final x and y co-ordinates of the Robot
INPUT:
x-2 y3 W3 N1
OUTPUT:
x-5 y4
"""
l=list(input().strip().split())
sumx,sumy=0,0
for i in range(len(l)):
    a=l[i]
    ch,s=a[0],str(a[1:])
    num=int(s)
    if ch=='x' or ch=='E':
        sumx=sumx+num
    elif ch=='W':
        sumx=sumx-num
    elif ch=='N' or ch=='y':
        sumy=sumy+num
    else:
        sumy=sumy-num
sumx=str(sumx)
sumy=str(sumy)
print("x"+sumx+" y"+sumy)

File: Robot co-ordinates.cpp
#include <bits/stdc++.h>

using namespace std;

int main(int argc, char** argv)
{
int sumx=0,sumy=0;
int num;
char ch,ch1;
while(scanf("%c%d",&ch,&num)>0)
{
    if(ch=='x'||ch=='E')
    {
        sumx=sumx+num;
    }
    else if(ch=='W')
    {
        sumx=sumx-num;
    }
    else if(ch=='N'||ch=='y')
    {
        sumy=sumy+num;
    }
    else if(ch=='S')
    {
        sumy=sumy-num;
    }
}
cout<<"x"<<sumx<<" "<<"y"<<sumy;



}
